packages/react-core/src/components/Label/__tests__/Label.test.tsx

```diff
@@ -109,4 +109,25 @@ describe('Label', () => {
     expect(screen.queryByRole('button', { name: 'Something' })).toBeNull();
     expect(asFragment()).toMatchSnapshot();
   });
+
+  test('clickable label', () => {
```

Comment:
Could you edit this test name to be a bit more descriptive? I.e. "a button is rendered when x is passed" or something along those lines.

Also I would like to see some "negative" tests here that assert things work as we expect in the opposite cases. For instance that your `expect(screen.getByRole(`button`)).toBeVisible();` fails if `onLabelClick` _isn't_ passed, and that the callback isn't called when it's passed but the label _isn't_ clicked.